server/routes/postRoutes.ts

```diff
@@ -2,84 +2,106 @@ import express = require('express');
+}
+
 })
 
 // Create post
-router.post('/', validatePost, (req, res) => {
-  // Post now using constructor with static id values. Once the project moves on to use db the id's can be auto incremented and fetched from the correct users.
-  console.log(req.body);
-  const post : Post = req.body;
-  posts.push(post)
-  res.status(201).send(post);
+// Now requires user id to make a post...
+router.post('/', validatePost, async (req, res, next) => {
+  try {
+    const user : User = plainToClass(User, await getRepository(User).findOne(req.body.ownerId));
+    if(user) {
+      const post : Post = new Post(user, req.body.category, req.body.message);
+      await getRepository(Post).save(post);
+      res.status(201).send(post);
+    }
+    else {
+      next(Boom.notFound("User doesn't exist"));
+    }
+  } catch (error) {
+    next(Boom.badImplementation());
+  }
 })
 
 // Delete post
-router.delete('/:id', (req, res) => {
-  const deleted: Post = posts[req.params.id];
-  // Not very optimal to leave holes with null, but splicing changes the positioning for the rest of the array.
-  // The problem should disapper when working with a database, with real id's and not indexes.
-
-  // Now compares objects and removes the matching one (same can be achieved with splice)
-  posts = posts.filter((post) => !comparePosts(post, deleted));
-  res.status(202).send(deleted);
-
+router.delete('/:id', validateParams, async (req, res, next) => {
+  try {
+    const deleted = await getRepository(Post).delete(req.params.id);
+    if(deleted.affected) {
+      res.sendStatus(204);
+    }
+    else {
+      next(notFound);
+    }
+  } catch (error) {
+    next(Boom.badImplementation());
+  }
 })
 
  // Update post, change the tittle and/or the message
- router.put('/:id', validatePostPUT, (req, res) => {
-   let updated: Post = posts[req.params.id];
-   if(req.body.category) {
-     updated.setCategory(req.body.category);
+ // LETS CHANGE ID
+ router.put('/:id', validateParams, validatePostPUT, async (req, res, next) => {
+   try {
+     await getRepository(Post).update(req.params.id, req.body);
+     const updated = await getRepository(Post).findOne(req.params.id);
+     if(updated) {
+       res.status(200).send(updated);
+     }
+     else {
+       next(notFound);
+     }
+   } catch (error) {
+     next(Boom.badImplementation());
    }
-   if(req.body.message) {
-     updated.setMessage(req.body.message);
-   }
-   posts[req.params.id] = updated;
-   res.status(200).send(updated);
  })
 
 
-// Like post
-router.patch('/:id/like', (req, res) => {
-  let updated: Post = posts[req.params.id];
-  updated.likePost();
-  res.status(200).send(updated);
-})
-
-// Don't like
-router.delete('/:id/like', (req, res) => {
-  let updated: Post = posts[req.params.id];
-  updated.unlikePost();
-  res.status(200).send(updated);
-})
 module.exports = router;
+
+
+/*
+  Skeleton of old sorting, f
+let responce = posts;
+  // parseInt to check if req.query.limit starts with a number
```

Comment:
Remove old code. No need to leave it commented because we are using version control so you will never lose them 🗡